pkg/gpio/gpio.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package gpio provides functions for interacting with GPIO pins via the
+// GPIO Sysfs Interface for Userspace.
+package gpio
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const gpioPath = "/sys/class/gpio"
+
+// Value represents the value of a gpio pin
```

Comment:
The only place I see these used is in SetGpioOutputValue below and there, you need a switch to get it to be a string. Would it make any sense at all to just make them have the const value "low" and "high"?

Further, it's two valued, might it just be a boolean? Were it a boolean, could you index an array? 

Or, could you just make a String() for value and then 
https://play.golang.org/p/bAqM6110dXf

Reply:
I wouldn't do a boolean, but I'd do a custom type like

```go
type Value string

const (
  Low Value = "low"
  High Value = "high"
)
```

or alternatively the String() method, but I'd prefer String() to be human-readable for printing, e.g.

```go
type Value bool

const (
  Low Value = false
  High Value = true
)

func (v Value) String() string {
  return fmt.Sprintf("GPIO Value %s", v.sysfsString())
}

func (v Value) sysfsString() string {
  if v {
    return "high"
  }
  return "low"
}
```

idk up to y'all

Reply:
I like the idea of having Value be a boolean to ensure its restricted to 0/1 - I'll try something along those lines

Reply:
Do we need to have a String() method?

Reply:
The problem I see with "GPIO Value %s" is that when we want to print a whole bunch of GPIOs as a vector (a common situation) that text may be a bit more than we need.

The nice thing about the bool is it is a nearly foolproof way to describe a variable with only two possible values, which is what we've got here. 

Bummer that go does not have subrange types but there we are ...

Reply:
The current version does what Chris' second suggestion does, but just prints "0" and "1" for String()